### PR TITLE
Fix windows builds

### PR DESCRIPTION
**Pull request by developit**
This fixes dev+prod builds on Windows.

I also added a type annotation to the webpack config that surfaced some required config updates we didn't make when upgrading to 4.x.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -53,18 +54,17 @@ module.exports = async function (_, env) {
...
     module: {
-      // Disable the default JavaScript handling:
-      defaultRules: [],
       rules: [
         {
           oneOf: [
             {
               test: /(\.mjs|\.esm\.js)$/i,
-              type: 'javascript/esm',
+              // don't use strict esm resolution for mjs:
```
Comment (RReverser) [inline webpack.config.js:66]: Why not?

**Comment by RReverser** [comment]
Awesome, thanks for working on this! Waiting for the CI to confirm :)

**Comment by RReverser** [comment]
Weird, Travis build is still not even created.

**Comment by RReverser** [comment]
I guess that was it - merging the dev unblocked Travis. Seems to work, thanks!

**Review by RReverser — APPROVED** [review]

File: webpack.config.js (hunk 1 above)
Comment (developit) [reply webpack.config.js:66]: Breaks a bunch of stuff in node_modules.

Comment (RReverser) [reply webpack.config.js:66]: Interesting. How come we didn't run into it before?

**Pull request merged**